docs/shaders.rst

```diff
@@ -0,0 +1,325 @@
+Writing PyGfx shaders
+=====================
+
+This document explains how to write shaders for PyGfx. This may be useful
+if you want to improve the existing shaders, add new shaders to PyGfx, or
+if you want to implement custom shaders in your own project.
+
+
+The shader class
+----------------
+
+A shader object derives from ``WorldObjectShader``. Its purposes is to
+provide (templated) shader-code (WGSL), set the corresponding template variables, define
+what bindings (buffers and textures) are used, and provide some more details
+of the pipeline and the rendering.
+
+The shader is associated with a WorldObject-material combination using the ``register_wgpu_render_function()``
+decorator. This decorator can be applied to your shader class, but it can also
+be applied to a function that returns multiple shader objects. This can be useful
+if you want multiple "passes", like a compute pass to prepare some data.
+
+The shader must implement a few methods. A typical shader is shown below:
+
+.. code-block:: python
+
+    from pygfx.renderers.wgpu import (
+        register_wgpu_render_function, WorldObjectShader, Binding, RenderMask
+    )
+
+    @register_wgpu_render_function(SomeWorldObject, SomeMaterial)
+    class SomeShader(WorldObjectShader):
+
+        type = "render"  # must be "render" or "compute"
+
+        def get_resources(self, wobject, shared):
+            bindings = {
+                0: Binding("u_stdinfo", "buffer/uniform", shared.uniform_buffer),
```

Comment:
Is `u_stdinfo` a mandatory binding?

Reply:
No, but it's common. I added the other 2 common uniforms to this doc (wobject's uniform buffer and material's uniform buffer).